tutorials/Python/neonUtilities/neonUtilitiesPython.md

```diff
@@ -0,0 +1,342 @@
+    
+    Unpacked  NEON.D08.TALL.DP1.10100.001.2014-07.expanded.20180308T182745Z.zip
+    
+    Unpacked  NEON.D01.HARV.DP1.10100.001.2014-07.expanded.20180308T182420Z.zip
+    
+    Unpacked  NEON.D05.UNDE.DP1.10100.001.2014-07.expanded.20180308T180317Z.zip
+    
+    Unpacked  NEON.D01.BART.DP1.10100.001.2014-08.expanded.20180308T175912Z.zip
+    
+    Finished: All of the data are stacked into  1  tables!
+    
+    Copied the first available NEON.University_of_Wyoming_Stable_Isotope_Facility.bgc_CNiso_externalSummary.csv to /stackedFiles
+    Copied the first available variable definition file to /stackedFiles and renamed as variables.csv
+    Copied the first available validation file to /stackedFiles and renamed as validation.csv
+    Stacked  sls_soilStableIsotopes
+    
+
+
+Check the folder containing the original zip file from the Data Portal; you should now have a subfolder containing the unzipped and stacked files.
+
+## Download files to be stacked
+
+The function `zipsByProduct()` uses the <a href="http://data.neonscience.org/data-api" target="_blank">NEON API</a> to programmatically download data files for a given product. The files downloaded by `zipsByProduct()` can then be fed into `stackByTable()`.
+
+The function will create a new folder in the R working directory and write the files there. Set the working directory if it isn't set to where you want it.
+
+
+```python
+base.setwd('/Downloads');
+```
+
+Run the downloader with these inputs: a DPID, a site (or "all" for all sites), a package (either basic or expanded), and an indicator to check the size of your download before proceeding, or not.
+
+There are two differences relative to running this function in R directly: 
+
+1. check.size becomes check_size, because dots have programmatic meaning in Python
+2. TRUE (or T) becomes "TRUE" because the values TRUE and FALSE don't have special meaning in Python the way they do in R, so it interprets them as variables if they're unquoted.
+
+check_size="TRUE" will estimate the size of the download and ask you to confirm before proceeding. This will cause problems in certain environments, or in batch processing. Under those circumstances, set check_size="FALSE", but consider the size of your query before doing this.
+
+
+```python
+neonUtils.zipsByProduct(dpID='DP1.10023.001', site='HARV', package='basic', check_size='TRUE');
+```
+
+    Continuing will download files totaling approximately 0.165245 MB. Do you want to proceed y/n: y
+    6 zip files downloaded to /Users/clunch/filesToStack10023
+    
+
+
+The message output by `zipsByProduct()` indicates the file path where the files have been downloaded.
+
+Now take that file path and pass it to `stackByTable()`. The file structure is slightly different from the zip file returned by the Portal, so we need an additional input, folder="TRUE".
+
+
+```python
+neonUtils.stackByTable("/Downloads/filesToStack10023", folder="TRUE");
```

Comment:
Need to add in the DPID. As written, this command gives me this error: 
`Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mjones01/anaconda3/lib/python3.6/site-packages/rpy2/robjects/functions.py", line 178, in __call__
    return super(SignatureTranslatedFunction, self).__call__(*args, **kwargs)
  File "/Users/mjones01/anaconda3/lib/python3.6/site-packages/rpy2/robjects/functions.py", line 106, in __call__
    res = super(Function, self).__call__(*new_args, **new_kwargs)
rpy2.rinterface.RRuntimeError: Error in (function (dpID, filepath, savepath = filepath, package = "basic",  : 
  ~/Downloads/filesToStack10023 is not a properly formatted data product ID. The correct format is DP#.#####.001, where the first placeholder must be between 1 and 4.`